src/Bicep.Core/Registry/LocalModuleRegistry.cs

```diff
@@ -16,41 +16,52 @@ namespace Bicep.Core.Registry
     public class LocalModuleRegistry : ModuleRegistry<LocalModuleReference>
     {
         private readonly IFileResolver fileResolver;
+        private readonly Uri parentModuleUri;
 
-        public LocalModuleRegistry(IFileResolver fileResolver)
+        public LocalModuleRegistry(IFileResolver fileResolver, Uri parentModuleUri)
         {
             this.fileResolver = fileResolver;
+            this.parentModuleUri = parentModuleUri;
         }
 
         public override string Scheme => ModuleReferenceSchemes.Local;
 
         public override RegistryCapabilities GetCapabilities(LocalModuleReference reference) => RegistryCapabilities.Default;
 
-        public override ModuleReference? TryParseModuleReference(string? alias, string reference, RootConfiguration configuration, out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder) =>
-            LocalModuleReference.TryParse(reference, out failureBuilder);
+        public override bool TryParseModuleReference(string? alias, string reference, [NotNullWhen(true)] out ModuleReference? moduleReference, [NotNullWhen(false)] out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
+        {
+            if (LocalModuleReference.TryParse(reference, parentModuleUri, out var @ref, out failureBuilder))
+            {
+                moduleReference = @ref;
+                return true;
+            }
+
+            moduleReference = default;
```

Comment:
It would be good to have a discussion with the team on use of `null` vs `default` for nullable types and see if we can get consensus on a convention either way.

Personally I prefer the explicitness of `null`:
* when reading/reviewing code using `default` you're forced to consider the possibility that you're dealing with a non-nullable value type (e.g. a record or an integer).
* it feels like a place where future bugs could slip in if you're not careful refactoring (e.g. refactoring a `myRecord?` to `myRecord` isn't going to cause any compiler errors).

Reply:
I ran into some odd behavior with `ImmutableArray` yesterday that moved me into the `null` camp. I had been using `default` as shorthand for "`null` or `new Nullable<T>()` when `T` is a value type" but did not realize structs could supply an alternative default value.